go/mcap/reader.go

```diff
@@ -141,29 +141,33 @@ func (r *Reader) Messages(
 	return r.unindexedIterator(ro.Topics, uint64(ro.Start), uint64(ro.End)), nil
 }
 
-func (r *Reader) readHeader() (*Header, error) {
-	_, err := r.rs.Seek(8, io.SeekStart)
-	if err != nil {
-		return nil, fmt.Errorf("failed to seek to header: %w", err)
+// Header reads the Header record from the MCAP.
+// NOTE: on non-seeking readers, this method will return an error if any other records
+// have been read from the MCAP file before.
```

Comment:
for a seeking reader, won't it also cause issues if other records have been read (i.e if you are in the middle of reading some stuff and call this?). Seems like you would lose your place in the file. Maybe worth a note in the docstring.

Reply:
we could also potentially scan forward to the Header in the constructor of the reader (it already reads the magic via the lexer - header should be right after). Then we could attach it to a Header field on the reader struct so you would be able to refer to it at any point during your reading. That might simplify some of the code here and reduce the amount of seeking around.

Reply:
it should also reduce the latency of Info() when calling against a remotely-hosted file. Currently the way that will work is

1) Open a reader
2) Read forward through the magic (in lexer constructor)
3) Issue a range request to exactly where currently located (offset=8)
4) Read the header
5) Open range request back to the summary section of the file and read that.

If the constructor for the reader read both the magic and the header (magic in the lexer, header in the reader constructor) then we would cut out step 3 and save a roundtrip.